src/jit/codegenxarch.cpp

```diff
@@ -1958,6 +1956,7 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
 #endif // !defined(_TARGET_64BIT_)
     case GT_ADD:
     case GT_SUB:
+        genConsumeOperands(treeNode->AsOp());
```

Comment:
Please add a note/assumption in function header of genCodeForBinary() that the caller of this method is expected to have called genConsumeOperands() before calling it.